### ISSUE TITLE
Getting instantaneous contacts between many pairs of bodies

### ISSUE THREAD

**Issue by josephrocca** [schteppe/cannon.js#216]
Hey @schteppe - this question is a somewhat related to [this issue](https://github.com/schteppe/cannon.js/issues/118). Basically I need to be able to check whether multiple pairs of bodies are in contact with each other at any instant

You showed us how to get the current contacts between two bodies:
```
var cg = new CANNON.ContactGenerator();
var bodyIndexList1 = [0], // Location of body 1 in the world.bodies array
    bodyIndexList2 = [1], // Location of body 2 in the world.bodies array
    result = [];
cg.getContacts(bodyIndex1,bodyIndex2,world,result);
// result is now an array of CANNON.ContactEquation
```
My first question is, if the `result` array is empty, that means there's no contact between the two bodies at that instant, right?

If so, is this the most efficient way of 'watching' the contacts between many pairs of bodies in cannonjs, or should I be using events like so:
```
   b1.addEventListener("collide", function(e) {
      if(e.with === b2) {
         //b1 and b2 are touching
      }
   });
```
I need to be able to test for contact at every instant but the `collide` event fires only once at the start of the contact, so unless there's something like a `contact` event, your code above using `CANNON.ContactGenerator` is the correct approach?

Thanks!

**Comment by schteppe** [comment]
To answer question 1, yes, if the array is empty then there were no contacts generated. (By the way, ContactGenerator has been renamed to Narrowphase).

Use only the Narrowphase if you are only interested in geometry and do not wish to use the physics. If you do use the physics (you do world.step() etc), then the World will automatically create contacts for you. Letting the World generate the contacts is probably the most efficient path, since it will use broadphase to skip unnecessary pairs.

If you feel like the contact event is not enough, then just check ```world.contacts``` after each step. This array contains ```ContactEquation```s with references to bodies and information about the contact. If you have a look at the Demo renderer, you'll see an example on how to render [contact points](https://github.com/schteppe/cannon.js/blob/e42557ab01aa1e62713f0b7f3d7e052f7405dd87/src/demo/Demo.js#L284) and [body-to-contact-point vectors](https://github.com/schteppe/cannon.js/blob/e42557ab01aa1e62713f0b7f3d7e052f7405dd87/src/demo/Demo.js#L300) (sorry for ugly code).

Regarding that "contact" event idea: yes, it would be nice to have. In p2.js I've implemented "beginContact" and "endContact" events, which is quite neat, but does not include contact events in between those... yet.

**Comment by josephrocca** [comment]
Ahh `world.contacts` - thanks! I appreciate the extra explanation too :)
